stablehlo/conversions/linalg/transforms/StablehloLegalizeToLinalg.cpp

```diff
@@ -1862,12 +1880,15 @@ struct GatherConversion final : OpConversionPattern<mlir::stablehlo::GatherOp> {
       indexFromOffset[remappedOffsetDim] = linalgIndices[offsetDim];
     }
 
-    // Now we add together our two indices to get the final index into the
+    // Now we add together our three indices to get the final index into the
     // operand.
     SmallVector<Value> combinedIndex;
     for (int64_t i = 0; i < operandRank; ++i)
       combinedIndex.push_back(rewriter.createOrFold<arith::AddIOp>(
-          loc, rewriter.getIndexType(), remappedIndexFromIndices[i],
+          loc, rewriter.getIndexType(),
+          rewriter.createOrFold<arith::AddIOp>(loc, rewriter.getIndexType(),
```

Comment:
I'm surprised this change in codegen didn't break any existing gather tests? If there are no batching dims on the gather op, will this generate a bunch of add 0's?